### PR TITLE
Add coroutine utill-test

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This adds a new module (`util-test`) which currently contains just a workaround which address [this issue](https://github.com/Kotlin/kotlinx.coroutines/issues/1996) since coroutines-test is not KMP.

## How is it being implemented?

It simply takes the implementation which was given in the mentioned issue. It also adds a 2 function (`runWithContextBlockingTest`), which overcomes a problem for integration testing and iOS (see test case).

Alongside it adds the debug release of `util`, to solve the current issue that debug is not resolving correctly on Android.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bitPogo** [comment]
Please note: Once settled on a name for the test module, the README will be updated.

File: util-test/src/androidMain/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  ~ Any usage of the SDK for other purposes, including usage for the development of
+  ~ applications/third-party applications shall require the conclusion of a license agreement
+  ~ between you and D4L.
+  ~
+  ~ If you are interested in licensing the SDK for your own applications/third-party
+  ~ applications and/or if you’d like to contribute to the development of the SDK, please
+  ~ contact D4L by email to help@data4life.care.
+  -->
+
+<manifest package="care.data4life.sdk.util" />
```
Comment (wmontwe) [inline util-test/src/androidMain/AndroidManifest.xml:17]: This should add `test` to the package

File: util-test/src/androidMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+package care.data4life.sdk.util.test
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.Executors
+import kotlin.coroutines.CoroutineContext
+
+actual val testCoroutineContext: CoroutineContext = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
```
Comment (wmontwe) [inline util-test/src/androidMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt:24]: spacing between the functions improves readability

File: util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package care.data4life.sdk.util.test
+
+import kotlinx.coroutines.CoroutineScope
+import kotlin.coroutines.CoroutineContext
+
+// see:https://github.com/Kotlin/kotlinx.coroutines/issues/1996
+expect val testCoroutineContext: CoroutineContext
```
Comment (wmontwe) [inline util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt:38]: spacing

File: util-test/src/commonTest/kotlin/care/data4life/sdk/util/test/BlockingTestTest.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+class BlockingTestTest {
+    @Test
+    fun `Given runBlocking is executed and contains CoroutineScope, it runs the given Scope`() = runBlockingTest {
+        // Given
+        val sample = "Hello CoroutineScope"
+        val channel = Channel<String>()
+
+        // When
+
```
Comment (wmontwe) [inline util-test/src/commonTest/kotlin/care/data4life/sdk/util/test/BlockingTestTest.kt:34]: too much spacing

File: util-test/src/iosMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package care.data4life.sdk.util.test
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+
+actual val testCoroutineContext: CoroutineContext = newSingleThreadContext("testRunner")
```
Comment (wmontwe) [inline util-test/src/iosMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt:39]: that's nice

**Review by wmontwe — CHANGES_REQUESTED** [review]
some small improvements could be done

File: util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt (hunk 3 above)
Comment (bitPogo) [reply util-test/src/commonMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt]: done

File: util-test/src/commonTest/kotlin/care/data4life/sdk/util/test/BlockingTestTest.kt (hunk 4 above)
Comment (bitPogo) [reply util-test/src/commonTest/kotlin/care/data4life/sdk/util/test/BlockingTestTest.kt]: done

File: util-test/src/androidMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt (hunk 2 above)
Comment (bitPogo) [reply util-test/src/androidMain/kotlin/care/data4life/sdk/util/test/BlockingTest.kt]: done

File: util-test/src/androidMain/AndroidManifest.xml (hunk 1 above)
Comment (bitPogo) [reply util-test/src/androidMain/AndroidManifest.xml]: done

**Review by wmontwe — APPROVED** [review]

**Pull request merged**